v4/source/groups.yaml

```diff
@@ -1,3 +1,563 @@
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+  '/groups/{group_id}/teams/{team_id}':
+    get:
+      tags:
+        - groups
+      summary: Get GroupSyncable from Team ID
+      description: |
+        Get the GroupSyncable object with group_id and team_id from params
+        ##### Permissions
+        Must have `manage_system` permission.
+
+        __Minimum server version__: 5.11
+      parameters:
+        - name: group_id
+          in: path
+          description: Group GUID
+          required: true
+          type: string
+        - name: team_id
+          in: path
+          description: Team GUID
+          required: true
+          type: string
+
+      responses:
+        '200':
+          description: GroupSyncable object retrieval successful
+          schema:
+            $ref: '#/definitions/GroupSyncableTeam'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+        '500':
+          $ref: '#/responses/InternalServerError'
+  
+  '/groups/{group_id}/channels/{channel_id}':
+    get:
+      tags:
+        - groups
+      summary: Get GroupSyncable from channel ID
+      description: |
+        Get the GroupSyncable object with group_id and channel_id from params
+        ##### Permissions
+        Must have `manage_system` permission.
+
+        __Minimum server version__: 5.11
+      parameters:
+        - name: group_id
+          in: path
+          description: Group GUID
+          required: true
+          type: string
+        - name: channel_id
+          in: path
+          description: Channel GUID
+          required: true
+          type: string
+
+      responses:
+        '200':
+          description: GroupSyncable object retrieval successful
+          schema:
+            $ref: '#/definitions/GroupSyncableChannel'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+        '500':
+          $ref: '#/responses/InternalServerError'
+
+  '/groups/{group_id}/teams':
+    get:
+      tags:
+        - groups
+      summary: Get group teams
+      description: |
+        Retrieve the list of teams associated to the group
+        ##### Permissions
+        Must have `manage_system` permission.
+
+        __Minimum server version__: 5.11
+      parameters:
+        - name: group_id
+          in: path
+          description: Group GUID
+          required: true
+          type: string
+
+      responses:
+        '200':
+          description: Teams list retrieval successful
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/GroupSyncableTeams'
```

Comment:
I think you can just do this...
```suggestion
              $ref: '#/definitions/GroupSyncableTeam'
```
... and then delete `GroupSyncableTeams` definition because this is returned as a simple array.

Reply:
I thought the same thing until I was playing around with endpoints. For example, when I hit the `groups/:group_id/teams/:team_id` endpoint, I get the following:  
![image](https://user-images.githubusercontent.com/17804942/67727709-ce330d00-f9c0-11e9-9f44-8ddb9713d349.png)

But when I hit `groups/:group_id/teams` endpoint, I get the following  
![image](https://user-images.githubusercontent.com/17804942/67727758-ffabd880-f9c0-11e9-8a4f-87de9b8bc6f3.png)

You can see a small discrepancy in the data returned (`team_display_name` and `team_type`) which I noticed which is why I created the two separate definitions. Should I omit the extra information and still revert it to a `GroupSyncableTeam`?

Reply:
Oh I see. Yes I had forgotten that there were extra fields. You're totally right those fields should be documented. Nice work.